docs/development/workflow/development_workflow.rst

```diff
@@ -619,6 +646,22 @@ The last command will open an editor with all your commits, allowing you to
 squash several commits together, rename them, etc. Helpfully, the file you are
 editing has the instructions on what to do.
 
+
+.. _github-squash-and-merge:
+
+Github 'Squash and merge'
```

Comment:
Moving this section to maintainer workflows page. Also going to add more info to it.